pages/instructorAssessmentSettings/instructorAssessmentSettings.js

```diff
@@ -0,0 +1,130 @@
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+const error = require('@prairielearn/prairielib/error');
+const logger = require('../../lib/logger');
+const { AssessmentCopyEditor, AssessmentRenameEditor, AssessmentDeleteEditor } = require('../../lib/editors');
+const { encodePath } = require('../../lib/uri-util');
+
+router.get('/', function(req, res, next) {
+    debug('GET /');
+    async.series([
+        function(callback) {
+            debug('query assessment_stats');
+            var params = {assessment_id: res.locals.assessment.id};
+            sqldb.queryOneRow(sql.assessment_stats, params, function(err, result) {
+                if (ERR(err, callback)) return;
+                res.locals.assessment_stat = result.rows[0];
+                debug(res.locals.assessment_stat);
+                callback(null);
+           });
+        },
+        function(callback) {
+            debug('query assessment_duration_stats');
+            // NOTE: this aggregates over all instances
+            var params = {assessment_id: res.locals.assessment.id};
+            sqldb.queryOneRow(sql.assessment_duration_stats, params, function(err, result) {
+                if (ERR(err, callback)) return;
+                res.locals.duration_stat = result.rows[0];
+                debug(res.locals.duration_stat);
+                callback(null);
+            });
+        },
+        (callback) => {
+            debug('query tids');
+            sqldb.queryOneRow(sql.tids, {course_instance_id: res.locals.course_instance.id}, (err, result) => {
+                if (ERR(err, callback)) return;
+                res.locals.tids = result.rows[0].tids;
+                callback(null);
+            });
+        },
+    ], function(err) {
+        if (ERR(err, next)) return;
+        debug('render page');
+        res.locals.infoAssessmentPath = encodePath(path.join('courseInstances', res.locals.course_instance.short_name, 'assessments', res.locals.assessment.tid, 'infoAssessment.json'));
+        res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+    });
+});
+
+router.post('/', function(req, res, next) {
+    debug('POST /');
+    if (req.body.__action == 'copy_assessment') {
+        debug('Copy assessment');
+        const editor = new AssessmentCopyEditor({
+            locals: res.locals,
+        });
+        editor.canEdit((err) => {
+            if (ERR(err, next)) return;
+            editor.doEdit((err, job_sequence_id) => {
+                if (ERR(err, (e) => logger.error(e))) {
+                    res.redirect(res.locals.urlPrefix + '/edit_error/' + job_sequence_id);
+                } else {
+                    debug(`Get assessment_id from tid=${editor.tid} with course_instance_id=${res.locals.course_instance.id}`);
+                    sqldb.queryOneRow(sql.select_assessment_id_from_tid, {tid: editor.tid, course_instance_id: res.locals.course_instance.id}, (err, result) => {
```

Comment:
Good point. As you say, it should be fine, but now I am a little paranoid because there could be weird things in the assessment set name (for example). So, I switched these calls to get the ID from a uuid instead of from a `tid`, `short_name`, or `qid`. Since the `uuid` is for sure created by us, we know it's valid, and there isn't a risk of SQL injection. I also added a comment as you suggested.

Reply:
in `fix_instance_switcher_bug`